src/eureka/S3_data_reduction/bright2flux.py

```diff
@@ -83,31 +83,36 @@ def dn2electrons(data, meta):
     nx = data.attrs['mhdr']['SUBSIZE1']
     ny = data.attrs['mhdr']['SUBSIZE2']
 
-    # Load gain array in units of e-/ADU
-    gain_header = fits.getheader(meta.gainfile)
-    xstart_gain = gain_header['SUBSTRT1']
-    ystart_gain = gain_header['SUBSTRT2']
+    if hasattr(meta, 'gain') and meta.gain is not None:
```

Comment:
If I want to specify meta.gain as a 2D array, would I have to write it out in the ECF or is there an easier way?

Reply:
At present yes, you'd have to manually enter the whole array in the ECF. The one thing I've thought of to allow easier manual input of array gains is allowing the user to set `meta.gainfile` instead of forcing users to use the current best file from CRDS. I could also add a check if the extension is .fits (as would be expected if using a CRDS gain file) or if the file extension is .csv (and then do some special processing). I don't think there is currently any demand for user-supplied 2D gain values though and allowing CSVs wouldn't be completely trivial (with complexities coming from subarrays), so I'd be more inclined to just allow the user to manually set `meta.gainfile` to a .fits file (and they can edit the FITS values if they want a fully custom array).